dotCMS/src/main/java/com/dotcms/util/DbExporterUtil.java

```diff
@@ -0,0 +1,321 @@
+                "- took: " + DateUtil.humanReadableFormat(
+                        Duration.of(
+                                System.currentTimeMillis() -starter,
+                                ChronoUnit.MILLIS)));
+
+        return file;
+    }
+
+    /**
+     * Resolves the path for pg_dump binary.
+     * It tries to find it by calling the `which pg_dump` command. If it cannot locate the command there, then copies it
+     * from the classpath to the dynamic content path.
+     */
+    static final Lazy<String> pgDumpPath = Lazy.of(() -> {
+        final Optional<String> pgDumpOpt = findExecutable("pg_dump");
+        if (pgDumpOpt.isPresent()) {
+            return pgDumpOpt.get();
+        }
+
+        final File pgDump = new File(ConfigUtils.getDynamicContentPath() + separator + "bin" + separator + "pg_dump");
+        if (!pgDump.exists()) {
+            copyPgDump(pgDump);
+        }
+
+        return pgDump.getAbsolutePath();
+    });
+
+    /**
+     * Copies pg_dumb binary to a defined folder for it to called later.
+     *
+     * @param pgDumpFile pg_dump binary file
+     */
+    @VisibleForTesting
+    static void copyPgDump(final File pgDumpFile) {
+        if (Files.notExists(pgDumpFile.toPath())) {
+            try {
+                Files.createDirectories(pgDumpFile.toPath());
+            } catch (IOException e) {
+                Logger.error(
+                        DbExporterUtil.class,
+                        String.format("Error creating directories %s", pgDumpFile.getAbsolutePath()),
+                        e);
+                throw new DotRuntimeException(e);
+            }
+        }
+
+        pgDumpFile.delete();
+
+        Logger.info(DbExporterUtil.class, "Copying pg_dump from: " + PG_ARCH.pgDump() + " to " + pgDumpFile);
+        try (final InputStream in = DbExporterUtil.class.getResourceAsStream("/" + PG_ARCH.pgDump());
+             final OutputStream out = Files.newOutputStream(pgDumpFile.toPath())) {
+            Logger.info(DbExporterUtil.class, String.format("In resource: %s", PG_ARCH.pgDump()));
+            Logger.info(DbExporterUtil.class, String.format("Out resource: %s", pgDumpFile.toPath()));
+
+            IOUtils.copy(in, out);
+        } catch (Exception e) {
+            Logger.error(
+                    DbExporterUtil.class,
+                    String.format("Could not copy PG dump for file: %s", pgDumpFile.getName()),
+                    e);
+            pgDumpFile.delete();
+            throw new DotRuntimeException(e);
+        }
+        pgDumpFile.setExecutable(true);
+    }
+
+    /**
+     * Calls the pg_dumb binary with necessary arguments and redirects standard output to {@link InputStream}.
+     *
+     * @return input stream with commands output
+     * @throws IOException
+     */
+    static InputStream exportSql() throws IOException {
+        final List<String> commands = getCommandAndArgs(pgDumpPath.get(), getDatasource().getDbUrl());
```

Comment:
The Command can be abstracted in a diff class so can be used on other context, rather than jus the DbExport